### PR TITLE
[iOS][Android] Update reanimated to `2.11.0`

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-6530.

# How

- Adjusted the vendored process to use the new version of the script on Android. 
- Run `et update-vendored-module -m react-native-reanimated` 
- Made sure that the integration between the reanimated and the gesture-handler still works 

# Test Plan

- bare-expo with NCL ✅

### CODE REVIEW

File: android/vendored/unversioned/react-native-reanimated/android/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+
```
Comment (Kudo) [inline android/vendored/unversioned/react-native-reanimated/android/README.md:1]: nit: remove this file from vendoring

File: android/vendored/unversioned/react-native-reanimated/android/rnVersionPatch/62/layoutReanimation/ReanimatedUIImplementation.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
+package com.facebook.react.uimanager;
```
Comment (Kudo) [inline android/vendored/unversioned/react-native-reanimated/android/rnVersionPatch/62/layoutReanimation/ReanimatedUIImplementation.java:1]: nit: remove the whole `rnVersionPatch` from vendoring. would be better we could apply the patch in the vendoring time.

File: tools/src/vendoring/config/expoGoConfig.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import path from 'path';
 
 import { Podspec } from '../../CocoaPods';
 import { EXPO_DIR, EXPOTOOLS_DIR } from '../../Constants';
+import { getExpoRepositoryRootDir, getReactNativeSubmoduleDir } from '../../Directories';
```
Comment (Kudo) [inline tools/src/vendoring/config/expoGoConfig.ts:7]:
```suggestion
import { EXPO_DIR, REACT_NATIVE_SUBMODULE_DIR } from '../../Constants';
```

also use the import names in the following code.

**Review by Kudo — COMMENTED** [review]
the _android/expoview/build.gradle_ is super clear now. great improvement for the new vendoring. i just leave some nit comments as following.

**Review by tsapeta — APPROVED** [review]
Solid work, thanks for moving reanimated to the new vendoring mechanism and clearing up expoview 👍

**Review by Kudo — COMMENTED** [review]
and also spotlessApply to fix android unit test failures

**Review by Kudo — APPROVED** [review]

**Pull request merged**